### PR TITLE
reference, toc: add a list of expressions pushed down

### PR DESCRIPTION
**Pull request by fzhedu**
### What is changed, added or deleted? 
Add a page that presents a list of expressions pushed down to TiKV

From the page, users can know which expressions have been pushed down to TiKV.

### What is the related PR or file link(s)? 
N/A

### Which version does your change affect? 
N/A

### CODE REVIEW

**Comment by zz-jason** [comment]
@lilin90 @CaitinChen PTAL

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+title: 下推到 TiKV 的表达式列表
+category: reference
+aliases: ['/docs-cn/sql/expressions-pushed-down-2-TiKV/']
+---
+
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion

# 下推到 TiKV 的表达式列表

```

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+---
+title: 下推到 TiKV 的表达式列表
+category: reference
+aliases: ['/docs-cn/sql/expressions-pushed-down-2-TiKV/']
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]: `aliases` is unnecessary for a newly added file. Please remove it.

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+---
+
+| 表达式分类 | 具体操作 |
+| -------------- | ------------------------------------- |
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
| :-------------- | :------------------------------------- |
```

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+| [逻辑运算](/reference/sql/functions-and-operators/operators.md) | AND (&&), OR (&#124;&#124;), NOT (!) |
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
| [逻辑运算](/reference/sql/functions-and-operators/operators.md#逻辑操作符) | AND (&&), OR (&#124;&#124;), NOT (!) |
```

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+| [比较运算](/reference/sql/functions-and-operators/operators.md) | <, <=, =, != (<>), >, >=, [`<=>`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#operator_equal-to), [`IN()`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in), IS NULL, LIKE, IS TRUE, IS FALSE, [`COALESCE()`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_coalesce) |
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
| [比较运算](/reference/sql/functions-and-operators/operators.md#比较方法和操作符) | <, <=, =, != (<>), >, >=, [`<=>`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#operator_equal-to), [`IN()`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in), IS NULL, LIKE, IS TRUE, IS FALSE, [`COALESCE()`](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_coalesce) |
```

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+| [JSON运算](/reference/sql/functions-and-operators/json-functions.md) | [JSON_TYPE(json_val)][json_type],<br> [JSON_EXTRACT(json_doc, path[, path] ...)][json_extract],<br> [JSON_UNQUOTE(json_val)][json_unquote],<br> [JSON_OBJECT(key, val[, key, val] ...)][json_object],<br> [JSON_ARRAY([val[, val] ...])][json_array],<br> [JSON_MERGE(json_doc, json_doc[, json_doc] ...)][json_merge], [JSON_SET(json_doc, path, val[, path, val] ...)][json_set], [JSON_INSERT(json_doc, path, val[, path, val] ...)][json_insert], [JSON_REPLACE(json_doc, path, val[, path, val] ...)][json_replace], [JSON_REMOVE(json_doc, path[, path] ...)][json_remove] |
+| [日期运算](/reference/sql/functions-and-operators/date-and-time-functions.md) | [`DATE_FORMAT()`](https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_date-format)  |
+
+注意：上述所支持下推的表达式可以被拉入黑名单mysql.expr_pushdown_blacklist中。
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
> **注意：**
>
> 上述支持下推的表达式可以被拉入黑名单 `mysql.expr_pushdown_blacklist` 中。
```

Comment (lilin90) [reply v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]: 什么时候可以或者有必要拉入黑名单呢？

**Comment by lilin90** [comment]
@fzhedu Does this new file also apply to `dev`, `v2.1`? If yes, please also update accordingly. Thanks!

**Comment by lilin90** [comment]
@fzhedu One of my comments is missed. Please check.

**Comment by XuHuaiyu** [comment]
@lilin90 What does this error message mean?
```
error: 1 files have invalid links (or alias links which are recommended to be replaced with latest one), please fix them!

=== ERROR REPORT == :

FILE: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md
[✖] /home/circleci/pingcap/docs-cn/v3.0//reference/sql/performance/understanding-the-query-execution-plan.md
Exited with code 1
```

**Comment by fzhedu** [comment]
> @lilin90 What does this error message mean?
> 
> ```
> error: 1 files have invalid links (or alias links which are recommended to be replaced with latest one), please fix them!
> 
> === ERROR REPORT == :
> 
> FILE: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md
> [✖] /home/circleci/pingcap/docs-cn/v3.0//reference/sql/performance/understanding-the-query-execution-plan.md
> Exited with code 1
> ```

the path is error.
 /home/circleci/pingcap/docs-cn/v3.0/reference/**sql/**performance/understanding-the-query-execution-plan.md

**Comment by fzhedu** [comment]
> @fzhedu Does this new file also apply to `dev`, `v2.1`? If yes, please also update accordingly. Thanks

already updated related files for `2.1`, but I don't know how to update `dev`

File: v2.1/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+|   └─Selection_7       | 3333.33  | cop  | gt(test.t.a, 2)                                            |
+|     └─TableScan_6     | 10000.00 | cop  | table:t, range:[-inf,+inf], keep order:false, stats:pseudo |
++-----------------------+----------+------+------------------------------------------------------------+
+4 rows in set (0.00 sec)
+```
+
+注：
+
+1. `admin reload expr_pushdown_blacklist` 只对执行该 SQL 的 TiDB server 生效，若需要集群中所有 TiDB server 生效，需要在每台 TiDB server 上执行该 SQL。
+2. 表达式黑名单功能在 v3.0.0 及以上版本中支持。
```
Comment (XuHuaiyu) [inline v2.1/reference/sql/functions-and-operators/expressions-pushed-down.md]: 2.1 do not support the expression blacklist.
So we do not need to add the related part in 2.1.

Comment (fzhedu) [reply v2.1/reference/sql/functions-and-operators/expressions-pushed-down.md]: ok, I will delete it.

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+4 rows in set (0.00 sec)
+```
+
+注：
+
+1. `admin reload expr_pushdown_blacklist` 只对执行该 SQL 的 TiDB server 生效，若需要集群中所有 TiDB server 生效，需要在每台 TiDB server 上执行该 SQL。
+2. 表达式黑名单功能在 v3.0.0 及以上版本中支持。
+3. 使用表达式原始名称文本（如，">"，"+"，"is null"）添加黑名单在 v3.0.3 及以下版本中不完全支持。v3.0.3 及以下版本中，部分表达式在黑名单中使用别名。已支持下推的表达式中，表达式别名与原始名不同的，对应关系见下表(不区分大小写)。
+
+| 表达式原始名称 | 表达式别名 | 备注 |
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]: If the 3rd column is empty, is it necessary to keep it? How about removing it?

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
++-----------------------+----------+------+------------------------------------------------------------+
+| Selection_5           | 2666.67  | root | lt(test.t.a, 2)                                            |
+| └─TableReader_8       | 3333.33  | root | data:Selection_7                                           |
+|   └─Selection_7       | 3333.33  | cop  | gt(test.t.a, 2)                                            |
+|     └─TableScan_6     | 10000.00 | cop  | table:t, range:[-inf,+inf], keep order:false, stats:pseudo |
++-----------------------+----------+------+------------------------------------------------------------+
+4 rows in set (0.00 sec)
+```
+
+注：
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
> **注意：**
```

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
>
```

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+1. `admin reload expr_pushdown_blacklist` 只对执行该 SQL 的 TiDB server 生效，若需要集群中所有 TiDB server 生效，需要在每台 TiDB server 上执行该 SQL。
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]: DO NOT use an ordered list if the order is not important.

```suggestion
> - `admin reload expr_pushdown_blacklist` 只对执行该 SQL 的 TiDB server 生效，若需要集群中所有 TiDB server 生效，需要在每台 TiDB server 上执行该 SQL。
```

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+2. 表达式黑名单功能在 v3.0.0 及以上版本中支持。
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
> - 表达式黑名单功能在 v3.0.0 及以上版本中支持。
```

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+3. 使用表达式原始名称文本（如，">"，"+"，"is null"）添加黑名单在 v3.0.3 及以下版本中不完全支持。v3.0.3 及以下版本中，部分表达式在黑名单中使用别名。已支持下推的表达式中，表达式别名与原始名不同的，对应关系见下表(不区分大小写)。
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]:
```suggestion
> - 使用表达式原始名称文本（如，">"，"+"，"is null"）添加黑名单在 v3.0.3 及以下版本中不完全支持。v3.0.3 及以下版本中，部分表达式在黑名单中使用别名。已支持下推的表达式中，表达式别名与原始名不同的，对应关系见下表（不区分大小写）。
```

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+| [JSON运算](/reference/sql/functions-and-operators/json-functions.md) | [JSON_TYPE(json_val)][json_type],<br> [JSON_EXTRACT(json_doc, path[, path] ...)][json_extract],<br> [JSON_UNQUOTE(json_val)][json_unquote],<br> [JSON_OBJECT(key, val[, key, val] ...)][json_object],<br> [JSON_ARRAY([val[, val] ...])][json_array],<br> [JSON_MERGE(json_doc, json_doc[, json_doc] ...)][json_merge],<br> [JSON_SET(json_doc, path, val[, path, val] ...)][json_set],<br> [JSON_INSERT(json_doc, path, val[, path, val] ...)][json_insert],<br> [JSON_REPLACE(json_doc, path, val[, path, val] ...)][json_replace],<br> [JSON_REMOVE(json_doc, path[, path] ...)][json_remove] |
+| [日期运算](/reference/sql/functions-and-operators/date-and-time-functions.md) | [`DATE_FORMAT()`](https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_date-format)  |
+
+## 禁止特定表达式下推
+
+当函数的计算过程由于下推而出现不被期待的行为时，我们通过将其拉入黑名单禁止下推来快速恢复业务。
+上述支持下推的表达式可以被拉入黑名单 `mysql.expr_pushdown_blacklist` 中，以禁止其下推。
+
+* 如何拉入黑名单？
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]: Please update the indentation. It's not correct. Indent sub-items with 4 spaces. You can check this rule in our style guide.

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 15]
```diff
@@ -0,0 +1,141 @@
+---
+title: 下推到 TiKV 的表达式列表
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]: Add a line of `summary`.

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+category: reference
+---
+
+# 下推到 TiKV 的表达式列表
+
```
Comment (lilin90) [inline v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]: Add a brief intro here to this whole doc.

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md (hunk 8 above)
Comment (fzhedu) [reply v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]: removed.

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md (hunk 15 above)
Comment (fzhedu) [reply v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]: added

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md (hunk 16 above)
Comment (fzhedu) [reply v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]: Added.

File: v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md (hunk 14 above)
Comment (fzhedu) [reply v3.0/reference/sql/functions-and-operators/expressions-pushed-down.md]: The indentation is added.

**Comment by lilin90** [comment]
@fzhedu I updated the organization and wording. Please check whether it's what you mean.

> > @fzhedu Does this new file also apply to `dev`, `v2.1`? If yes, please also update accordingly. Thanks
> 
> already updated related files for `2.1`, but I don't know how to update `dev`

- Could you please tell me where did you updated `v2.1` because I didn't see changes.
- To update `dev`, add changes to `pingcap/docs-cn/dev` like that for `v3.0`.

**Comment by fzhedu** [comment]
> @fzhedu I updated the organization and wording. Please check whether it's what you mean.
> 
> > > @fzhedu Does this new file also apply to `dev`, `v2.1`? If yes, please also update accordingly. Thanks
> > 
> > 
> > already updated related files for `2.1`, but I don't know how to update `dev`
> 
> * Could you please tell me where did you updated `v2.1` because I didn't see changes.
> * To update `dev`, add changes to `pingcap/docs-cn/dev` like that for `v3.0`.



> @fzhedu I updated the organization and wording. Please check whether it's what you mean.
> 
> > > @fzhedu Does this new file also apply to `dev`, `v2.1`? If yes, please also update accordingly. Thanks
> > 
> > 
> > already updated related files for `2.1`, but I don't know how to update `dev`
> 
> * Could you please tell me where did you updated `v2.1` because I didn't see changes.
> * To update `dev`, add changes to `pingcap/docs-cn/dev` like that for `v3.0`.

The changes in `v2.1` is removed because the blacklist is supported since `v3.0`, which is verified by @XuHuaiyu  .
Is the version of `dev` is larger than `v3.0`?

**Comment by lilin90** [comment]
@fzhedu Yes. `dev` is for the future `v4.0`.

**Pull request merged**